### PR TITLE
Multiple improvements for comment edit diff

### LINKED ISSUE #21924: Scale of avatar for user in Issue history is larger than it should be
main branch, https://try.gitea.io , any issue with edit history, eg: https://try.gitea.io/wxiaoguang/test/issues/1

The avatar is broken, it's shown as the original size.

image

### PR DESCRIPTION
**Pull request by silverwind**
- Use explicit avatar size so when JS copies the HTML, the size gets copied with it
- Replace icon font use with SVG
- Improve styling and diff rendering
- Sort lists in `svg.js`

Fixes: https://github.com/go-gitea/gitea/issues/21924

Screenshot 2022-11-30 at 17 52 17

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]

File: routers/web/repo/issue_content_history.go [hunk 1]
```diff
@@ -63,16 +64,20 @@ func GetContentHistoryList(ctx *context.Context) {
...
+		src := html.EscapeString(item.UserAvatarLink)
+		class := avatars.DefaultAvatarClass + " mr-3"
+		name := html.EscapeString(username)
+		avatarHTML := string(templates.AvatarHTML(src, 28, class, username))
+		timeSinceText := string(timeutil.TimeSinceUnix(item.EditedUnix, ctx.Locale))
+
 		results = append(results, map[string]interface{}{
-			"name": fmt.Sprintf("<img class='ui avatar image' src='%s'><strong>%s</strong> %s %s",
-				html.EscapeString(item.UserAvatarLink), html.EscapeString(username), actionText, timeSinceText),
+			"name":  avatarHTML + "<strong>" + name + "</strong> " + actionText + " " + timeSinceText,
```
Comment (delvh) [inline routers/web/repo/issue_content_history.go:80]: I don't quite understand this change:
Why is it no longer a `fmt.Sprintf` but a simple string concatenation?

**Review by lafriks — APPROVED** [review]

File: web_src/js/features/repo-issue-content.js [hunk 2]
```diff
@@ -79,7 +76,7 @@ function showContentHistoryMenu(issueBaseUrl, $item, commentId) {
   const $headerLeft = $item.find('.comment-header-left');
   const menuHtml = `
   <div class="ui pointing dropdown top left content-history-menu" data-comment-id="${commentId}">
-    <a>&bull; ${i18nTextEdited} ${svg('octicon-triangle-down', 17)}</a>
+    &bull; <a>${i18nTextEdited} ${svg('octicon-triangle-down', 14, 'dropdown icon')}</a>
```
Comment (delvh) [inline web_src/js/features/repo-issue-content.js]: Hmm, should the bullet really not be part of the link anymore?

Comment (wxiaoguang) [reply web_src/js/features/repo-issue-content.js]: IIRC, in the first PR for the history, there was no underline (when hover)

Some later PRs make the underline appear, so here the `<a>` needs a `muted`, then no need to move the bullet out (or it's also fine to move it out, either is fine)

File: routers/web/repo/issue_content_history.go (hunk 1 above)
Comment (wxiaoguang) [reply routers/web/repo/issue_content_history.go:80]: IMO string concatenation is also clear enough

Comment (silverwind) [reply routers/web/repo/issue_content_history.go:80]: I prefer concatenation as it is more readable to me. A shame golang does not have better string interpolation.

File: web_src/js/features/repo-issue-content.js (hunk 2 above)
Comment (silverwind) [reply web_src/js/features/repo-issue-content.js]: `muted` only changes color, not underline. That said, I think only the "edited" should be underlined, but currently the whitspace after also is.

Comment (silverwind) [reply web_src/js/features/repo-issue-content.js]: Updated:

image

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
It flicks if the network is not fast.


The first loading UI doesn't look good.


image

File: web_src/js/features/repo-issue-content.js [hunk 3]
```diff
@@ -13,20 +13,17 @@ function showContentHistoryDetail(issueBaseUrl, commentId, historyId, itemTitleH
...
+    <div>${itemTitleHtml}</div>
+    <div class="ui dropdown dialog-header-options df ac mr-5 hide">
+      ${i18nTextOptions}${svg('octicon-triangle-down', 14, 'dropdown icon')}
       <div class="menu">
         <div class="item red text" data-option-item="delete">${i18nTextDeleteFromHistory}</div>
       </div>
     </div>
   </div>
-  <!-- ".modal .content" style was polluted in "_base.less": "&.modal > .content"  -->
-  <div class="scrolling content" style="text-align: left; min-height: 30vh;">
```
Comment (wxiaoguang) [inline web_src/js/features/repo-issue-content.js:27]: IMO The `min-height: 30vh` was done by purpose:
* to show the loading UI without flicker
* to make the dialog look better if there are only a few lines in the diff.

Comment (silverwind) [reply web_src/js/features/repo-issue-content.js:27]: We could just eliminate the loading state of the modal and fetch all data after click and only show the modal when all data is present.

I don't agree about superflous whitespace on small diffs 😉

Comment (wxiaoguang) [reply web_src/js/features/repo-issue-content.js:27]: The user needs a UI feed back when they clicks the history menu item, so, something must be shown.


The dialog height in current approach can be very small, which makes the loading UI look strange.

Comment (silverwind) [reply web_src/js/features/repo-issue-content.js:27]: > The user needs a UI feed back when they clicks the history menu item, so, something must be shown.

Could probably add a `is-loading` spinner onto the clicked dropdown item. A bit unconventional but should work.

> The dialog height in current approach can be very small, which makes the loading UI look strange.

Please show an example.

Comment (wxiaoguang) [reply web_src/js/features/repo-issue-content.js:27]: The example was there: https://github.com/go-gitea/gitea/pull/21990#issuecomment-1335085467

Or you can reproduce by:

* open the history menu
* disconnect your network (or emulate a laggy network)
* click a history item

Comment (silverwind) [reply web_src/js/features/repo-issue-content.js:27]: Yes that's the broken loading state, but if we don't have a loading state inside the modal, this won't be an issue. I'll check later.

**Pull request merged**